apache/superset
# feat(plugin-chart-echarts): [feature-parity] support extra control for the area chart V2

🏆 Enhancements

This PR implements `Extra Control` on the area chart V2, and the chart's update mechanism is using `setControlValue` hook. There are some differences or improvements:
- In explore the stack control will be updated if `Extra Control` is updated. But for NVD3, updating extra control does not update the stack control. 
- Here only keep  `Stack` and `Expend` two options (Stream Graph will come soon) and add a new `None` option, which can cancel the stack style.
- Another UI difference is the extra control button which uses the same radio control as the control panel instead of NVD3 radio button.

## explore

### before

https://user-images.githubusercontent.com/11830681/167296244-ea5d6bd5-f7b6-4abc-a058-d9fd9e894242.mov




### after


https://user-images.githubusercontent.com/11830681/167295413-aa4379a8-8442-4342-90d6-fff81a638a4a.mov




## dashboard

### before

https://user-images.githubusercontent.com/11830681/167296261-d5450c9f-c00d-421c-ad7a-f29368e266cd.mov





### after


https://user-images.githubusercontent.com/11830681/167296426-05a99519-fcf6-43df-8639-ec4692ea6442.mov





### TESTING INSTRUCTIONS

1. add `Exta Control` to the time-series area chart
2. save and go to dashboard
3. test the control and see results

### ADDITIONAL INFORMATION


- [x] Has associated issue: https://github.com/apache/superset/issues/16298
